lib/webview.dart

```diff
@@ -92,47 +117,43 @@ class _ScreenerAppState extends State<ScreenerApp> {
-                    });
-                  }
-                  if (url.contains('premium')) {
-                    await controller.runJavascript(
-                        "if (document.getElementById('razorpay-info')){var info = document.getElementById('razorpay-info'); btn1 = document.createElement('button');function cloneAttributes(element, sourceNode) { let attr; let attributes = Array.prototype.slice.call(sourceNode.attributes); while(attr =attributes.pop()) {element.setAttribute(attr.nodeName, attr.nodeValue);}};cloneAttributes(btn1, info); info.parentElement.append(btn1);info.style.display = 'none'; btn1.innerText='BUY NOW';btn1.addEventListener('click', function() {options = {'key': info.getAttribute('data-key'),'amount': info.getAttribute('data-amount'),'currency': 'INR','name': 'Mittal Analytics (P) Ltd','description': info.getAttribute('data-description'),'display_currency': info.getAttribute('data-display_currency'),'display_amount': info.getAttribute('data-display_amount'),'prefill': {'name': info.getAttribute('data-prefill.name'),'email': info.getAttribute('data-prefill.email')},'handler': function (response) {var inputs = info.form.elements;for (var i = 0; i < inputs.length; i++) {if (inputs[i].name === 'razorpay_payment_id') {inputs[i].value = response.razorpay_payment_id}};info.form.submit()},'notes': {'plan_name': info.getAttribute('data-notes.plan_name'),'user_id': info.getAttribute('data-notes.user_id')}};RAZORPAY.postMessage(JSON.stringify(options))})}");
-                  }
-                },
-                // ignore: prefer_collection_literals
-                javascriptChannels: <JavascriptChannel>[
-                  _razorpayChannel(),
-                ].toSet(),
-              ),
-              isLoading
-                  ? const Center(child: CircularProgressIndicator())
-                  : Stack(),
-            ]),
+            body: WebView(
+              initialUrl: _screenerHomeUrl,
+              javascriptMode: JavascriptMode.unrestricted,
+              userAgent: _proxyUserAgent,
+              onWebViewCreated: (controller) {
+                this.controller = controller;
+              },
+              zoomEnabled: false,
+              navigationDelegate: (NavigationRequest request) {
+                if (request.url.contains('premium')) {
+                  return NavigationDecision.navigate;
+                } else if (request.url.startsWith(_screenerHomeUrl)) {
+                  return NavigationDecision.navigate;
+                } else if (request.url.contains("google")) {
+                  return NavigationDecision.navigate;
+                } else {
+                  _launchURL(request.url);
+                  return NavigationDecision.prevent;
+                }
+              },
+              onPageFinished: (String url) async {
+                if (url.contains('premium')) {
+                  await controller.runJavascript("if (document.getElementById('razorpay-info'))" +
```

Comment:
JS can be formatted with more line breaks for better readability. Not sure if we should do this though as the file may get too large then.